### PR TITLE
New Resource: `azurerm_arc_machine_extension`

### PR DESCRIPTION
**Pull request by liuwuliuyun**
New Resource: `azurerm_arc_machine_extension`
Swagger: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/hybridcompute/resource-manager/Microsoft.HybridCompute/stable/2022-11-10/HybridCompute.json

Testing evidence will be provided after go test run completes.

### CODE REVIEW

File: website/docs/r/arc_machine_extension.html.markdown [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+* `name` - (Required) The name which should be used for this Hybrid Compute Machine Extension. Changing this forces a new Hybrid Compute Machine Extension to be created.
+
+* `publisher` - (Required) The name of the extension handler publisher, such as `Microsoft.Azure.Monitor`. Changing this forces a new Hybrid Compute Machine Extension to be created.
+
+* `type` - (Required) Specifies the type of the extension. For example `CustomScriptExtension` or `AzureMonitorLinuxAgent`. Changing this forces a new Hybrid Compute Machine Extension to be created.
+
+---
+
+* `automatic_upgrade_enabled` - (Optional) Indicates whether the extension should be automatically upgraded by the platform if there is a newer version available. Supported values are `true` and `false`.
```
Comment (katbyte) [inline website/docs/r/arc_machine_extension.html.markdown:51]: grammer
```suggestion
* `automatic_upgrades_enabled` - (Optional) Indicates whether the extension should be automatically upgraded by the platform if there is a newer version available. Supported values are `true` and `false`.
```

**Review by katbyte — CHANGES_REQUESTED** [review]
two minor comments but otherwise looks good

File: website/docs/r/arc_machine_extension.html.markdown (hunk 1 above)
Comment (liuwuliuyun) [reply website/docs/r/arc_machine_extension.html.markdown]: @katbyte I'd love to change the property name to suggested. But I found similar property named `automatic_upgrade_enabled` in [azurerm_virtual_machine_extension](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_machine_extension#automatic_upgrade_enabled). Which one should I use?

Comment (katbyte) [reply website/docs/r/arc_machine_extension.html.markdown]: ah lets keep it wrong for consistency then

**Pull request merged**